Add Trainer.validate(…) method to run one validation epoch (#4707)
PyTorchLightning/pytorch-lightning: docs/source/trainer.rst

------------

Validation
----------
After training, you can perform a new evaluation epoch over the validation set
with :meth:`pytorch_lightning.trainer.trainer.Trainer.validate`. This might be

> useful if you want to collect new metrics from a model you already trained.

Reviewer: only possible after training?

Author: Not necessarily, indeed. What I wanted to clarify is that you don't need this method in the normal training loop, it's only for performing validation on its own. Maybe this is better?

```suggestion
You can perform an evaluation epoch over the validation set, outside of the training loop,
using :meth:`pytorch_lightning.trainer.trainer.Trainer.validate`. This might be
useful if you want to collect new metrics from a model that has already been trained.
```

Reviewer: updated a bit. Might solve this issue https://github.com/PyTorchLightning/pytorch-lightning/issues/4170

Author: Wonderful
